components/raftstore/src/store/peer.rs

```diff
@@ -2117,10 +2122,11 @@ impl Peer {
                 // before or after the previous read index, and the lease can be renewed when get
                 // heartbeat responses.
                 LeaseState::Valid | LeaseState::Expired => {
+                    let committed_index = self.get_store().committed_index();
                     if let Some(read) = self.pending_reads.back_mut() {
                         let max_lease = poll_ctx.cfg.raft_store_max_leader_lease();
                         if read.renew_lease_time + max_lease > renew_lease_time {
-                            read.push_command(req, cb);
+                            read.push_command(req, cb, committed_index);
```

Comment:
Suppose there are 2 read index requests A and B, and A reaches the TiKV peer earlier. Seems B can get a less index than A. Is it correct?

Reply:
It doesn't matter. These two operations are parallel so the order is not guaranteed.